packages/lib/server/repository/membership.ts

```diff
@@ -0,0 +1,108 @@
+import { prisma } from "@calcom/prisma";
+import type { Prisma, MembershipRole } from "@calcom/prisma/client";
+
+import logger from "../../logger";
+import { safeStringify } from "../../safeStringify";
+import { LookupTarget, ProfileRepository } from "./profile";
+
+const log = logger.getSubLogger({ prefix: ["repository/membership"] });
+type IMembership = {
+  teamId: number;
+  userId: number;
+  profileId: number | null;
+  accepted: boolean;
+  role: MembershipRole;
+};
```

Comment:
My goal was to not use Prisma type for repository methods so that we can replace Prisma with any other ORM when needed with least friction possible. 

Though I have still used it at some places, but that's because I was saving some time. Wherever I though I could quickly get it done without Prisma types, I chose to do that like here.

Reply:
That makes total sense. I think in future PRs I'll start taking this approach as well.